### ISSUE TITLE
Possibility of encoding binary data in Aztec code?

### ISSUE THREAD

**Issue by Median555** [micjahn/ZXing.Net#149]
Hi!

After doing some Googling, it seems that Aztec codes should be able to contain pure binary data (not encoded in a container format like base64). However, it only seems possible to give a string to a `BarcodeWriter`.

Is it possible somehow to just give it a byte array instead, making it read from that?

I tried the following code (to make it only read binary data), but I'm basically guessing and the outputted barcodes are not correct:

```cs
public Bitmap GenerateAztec(byte[] data)
{
	AztecCode code = ZXing.Aztec.Internal.Encoder.encode(data);

	BitmapRenderer renderer = new BitmapRenderer();

	return renderer.Render(code.Matrix, BarcodeFormat.AZTEC, content: null);
}
```

**Comment by micjahn** [comment]
There is no native support for byte data. But it should be possible with encoding byte data as ISO 8859-15.
```
var writer = new BarcodeWriter
{
    Format = BarcodeFormat.AZTEC,
};
var originalByteData = new byte[256];
var content = System.Text.Encoding.GetEncoding("ISO_8859-15").GetString(originalByteData);
var barcode = writer.Write(content);
```

**Comment by Median555** [comment]
Thank you for answering.

 I apologise for the super late reply, this question lead down quite the rabbit hole! 

I tried using the encoding you suggest, and indeed, it is working as intended :tada:  I actually tried all 140 encodings available on my computer and the one you suggested was the only one that worked.

However, I also tried decoding Aztec codes with binary data and discovered that it doesn't work currently. I did however find a fix, but will keep that discussion to another issue. I also discovered and issue with one of the mode tables. The fix for that is in pull request https://github.com/micjahn/ZXing.Net/pull/175.
